File: misc/versus/ruby/blastoff.rb
#-  Ruby source code

#-  blastoff.rb ~~
#                                                       ~~ (c) SRW, 18 Aug 2011

lambda { |argv|

    def countdown(n)
        (n <= 0) ? "Blastoff!" : n.to_s + " ... " + countdown(n - 1)
    end

    puts countdown(10)

}.call(ARGV)

#-  vim:set syntax=ruby:

File: misc/versus/javascript/blastoff.js
//- JavaScript source code

//- blastoff.js ~~
//                                                      ~~ (c) SRW, 09 Aug 2011

chassis(function (q) {
    "use strict";

    var countdown;

    countdown = function (n) {
        return (n <= 0) ? "Blastoff!" : n + " ... " + countdown(n - 1);
    };

    q.puts(countdown(10));

});

//- vim:set syntax=javascript:
